lib/blocktree/blocktree.go

```diff
@@ -19,187 +19,168 @@ package blocktree
 import (
 	"fmt"
 	"math/big"
+	"time"
 
 	"github.com/ChainSafe/gossamer/dot/core/types"
 	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/database"
-
-	log "github.com/ChainSafe/log15"
 	"github.com/disiqueira/gotree"
 )
 
 // Hash common.Hash
 type Hash = common.Hash
 
-// Database is the blocktree database
-type Database struct {
-	Db database.Database
-}
-
 // BlockTree represents the current state with all possible blocks
 type BlockTree struct {
-	head            *Node
-	leaves          leafMap
-	finalizedBlocks []*Node
-	Db              *Database
+	head   *node // genesis node
+	leaves leafMap
+	db     database.Database
+}
+
+// NewEmptyBlockTree creates a BlockTree with a nil head
+func NewEmptyBlockTree(db database.Database) *BlockTree {
+	return &BlockTree{
+		head:   nil,
+		leaves: make(leafMap),
+		db:     db,
+	}
 }
 
 // NewBlockTreeFromGenesis initializes a blocktree with a genesis block.
 // Currently passes in arrival time as a parameter instead of setting it as time of instanciation
-func NewBlockTreeFromGenesis(genesis types.Block, db *Database) *BlockTree {
-	head := &Node{
-		hash:        genesis.Header.Hash(),
-		number:      genesis.Header.Number,
+func NewBlockTreeFromGenesis(genesis *types.Header, db database.Database) *BlockTree {
+	head := &node{
+		hash:        genesis.Hash(),
 		parent:      nil,
-		children:    []*Node{},
+		children:    []*node{},
 		depth:       big.NewInt(0),
-		arrivalTime: genesis.GetBlockArrivalTime(),
+		arrivalTime: uint64(time.Now().Unix()), // TODO: genesis block doesn't need an arrival time, it isn't used in median algo
```

Comment:
issue number? can you explain?

Reply:
will address in #567 

arrivalTime is only relevant for blocks that we receive from other nodes on the network (the purpose of arrivalTime is to allow us to calculate what slot other nodes are on) since the genesis block comes from the genesis file, not the network, it's not relevant to the slot time calculation